code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -526,8 +526,10 @@ var/list/gaslist_cache = null
 	var/list/cached_gases = gases
 
 	for(var/id in cached_gases | sample_gases) // compare gases from either mixture
-		var/gas_moles = cached_gases[id] ? cached_gases[id][datatype] : 0
-		var/sample_moles = sample_gases[id] ? sample_gases[id][datatype] : 0
+		var/gas_moles = cached_gases[id]
+		gas_moles = gas_moles ? gas_moles[datatype] : 0

```

Comment:
potential optimisation:

```gas_moles = cached_gases[id] || 0```
this is a one liner version of what you have now, but I'm unsure if ```||``` short-circuiting is faster than a ternary op.